tests/test_haas_interface.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, Dict, Tuple
+import copy
+import random
+import pickle
+import string
+import time
+import threading
+import unittest
+import zmq
+
+from server.machine_interfaces import HaasInterface
+from lf_utils import retry
+
+class MockHaasInterface(HaasInterface):
+    def __init__(self, *args, **kwargs):
+        threading.Thread.__init__(self)
+        super().__init__(*args, **kwargs)
+
+        self._return_haas_data = False
+        self._wait_interval = 0.01
+
+    def set_haas_data(self, haas_data: Dict[str, Any]):
+        self._mock_haas_data = haas_data
+        self._return_haas_data = True
+    
+
+class HaasInterfaceTests(unittest.TestCase):
```

Comment:
this isn't testing the code you wrote at all. you need to be testing the poll machine code, as that is the only new thing that the inherited `HaasInterface` class introduces